internal/clients/imagebuilder/image_builder_client.go

```diff
@@ -0,0 +1,80 @@
+package imagebuilder
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+)
+
+const URL = "https://console.redhat.com/api/image-builder/v1/compose"
+
+func RequestImageBuilder(client *http.Client, endpoint string, method string, orgId string, body *strings.Reader) ([]byte, error) {
+	var req *http.Request
+	var err error
+
+	req, err = http.NewRequest(method, endpoint, body)
+	if err != nil {
+
+		return nil, err
+
+	}
+
+	req.Header.Set("x-rh-identity", orgId)
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	responseBody, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	return responseBody, nil
+}
+
+func CreateAMI(HTTPclient *http.Client, orgId string) error {
+	data, err := createAMIrequest()
+	if err != nil {
+		return err
+	}
+
+	body := strings.NewReader(data)
+	req, err := RequestImageBuilder(HTTPclient, URL, "POST", orgId, body)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println("the AMI's ID is:", string(req))
+	return nil
+}
+
+func createAMIrequest() (string, error) {
+	data := AmiResponse{
+		Distribution: "rhel-84",
```

Comment:
So this is a quirk on our end, but the non-GA images shouldn't really be used as they're not EUS (long term support) images and aren't updated anymore. They're essentially frozen-in-time rhel 8. For technical reasons we don't have rolling rhel-8/9 distros, but we could add the alias in the image-builder layer. 

This should still build of course, but as soon as you run dnf update you'll get rhel-86 (current GA) and there will be a lot to update.